Documentation/network/concepts/ipam/multi-pool.rst

```diff
@@ -0,0 +1,153 @@
+  kind: CiliumPodIPPool
+  metadata:
+    name: green-pool
+  spec:
+    ipv4:
+      cidrs:
+        - 10.20.0.0/16
+        - 10.30.0.0/16
+      maskSize: 24
+    ipv6:
+      cidrs:
+        - fd00::/104
+      maskSize: 120
+
+New pools can be added at run-time. The list of CIDRs in each pool can also be
+extended at run-time. In-use CIDRs may not be removed from an existing pool, and
+existing pools may not be deleted if they are still in use by a Cilium node.
+The mask size of a pool is immutable and the same for all nodes.
+
+Configuration
+*************
+
+Multi-Pool IPAM can be enabled using the ``ipam.mode=multi-pool`` Helm value.
+To have the Cilium operator automatically create ``CiliumPodIPPools`` custom
+resources at startup, use the ``ipam.operator.autoCreateCiliumPodIPPools`` Helm
+value. It contains a map which follows the ``CiliumPodIPPools`` CRD schema
+described above.
+
+.. code-block:: yaml
+
+  ipam:
+    mode: multi-pool
+    operator:
+      autoCreateCiliumPodIPPools:
+        default:
+          ipv4:
+            cidrs:
+              - 10.10.0.0/16
+            maskSize: 24
+        other:
+          ipv4:
+            cidrs:
+              - 10.20.0.0/16
+            maskSize: 24
+
+.. note::
+
+  For a practical tutorial on how to enable this mode in Cilium, see
+  :ref:`gsg_ipam_crd_multi_pool`.
+
+Allocation Parameters
+---------------------
+
+Cilium agent can be configured to pre-allocate a minimum number of IPs from each
+pool. This behavior can be controlled using the
+``ipam-multi-pool-pre-allocation`` flag.  It contains a key-value map of the
+form ``<pool-name>=<preAllocIPs>`` where ``preAllocIPs`` specifies how many IPs
+(both IPv4 and IPv6) are to be pre-allocated to the local node.
+
+The flag defaults to ``default=8``, which means it will always pre-allocate at least
+8 IPs from the ``default`` pool. All other pools which do not have an entry in the
+``ipam-multi-pool-pre-allocation`` map are assumed to have a ``preAllocIPs`` of zero,
+i.e. no IPs are pre-allocated for that pool.
+
+The formula Cilium agent uses to compute the number of needed IPs each each pool is:
+
+.. code-block:: go
+
+  neededIPs = roundUp(inUseIPs + pendingIPs + preAllocIPs, preAllocIPs)
+
+Where ``inUseIPs`` is the number of IPs that are currently actively in use,
+``pendingIPs`` number of IPs that have a pending pod (i.e. pods which have been
+scheduled on the node, but not yet received an IP), and ``preAllocIPs`` is the
+minimum number of IPs that we want to pre-allocate as a buffer, i.e. the value
+taken from the ``ipam-multi-pool-pre-allocation`` map.
+
+ .. _ipam_crd_multi_pool_limitations:
+
+Limitations
+***********
+
+Multi-Pool IPAM is a preview feature. The following limitations apply to Cilium running in
+``multi-pool`` IPAM mode:
+
+.. warning::
+   - Tunnel mode is not supported. Multi-Pool IPAM may only be used in direct routing mode.
+   - Transparent encryption with IPSec is not supported.
```

Comment:
This is part of the "Limitations" section. AFAIK there are no known limitations with Wireguard, so I'm not sure what we want to add here re. Wireguard?

Reply:
Ah I see. I think it would still be useful if we add a note making it clear that Wireguard is fully supported, even if IPSec isn't. Something like: 

```Transparent encryption is only supported with Wireguard and cannot be used with IPSec.```

Reply:
Changed to "Transparent encryption is only supported with Wireguard and cannot be used with IPSec."